api/app.py

```diff
@@ -143,6 +178,54 @@ def predict():
     # return cats[text_clf.predict([text])[0]]
     return cats[gs_clf.predict([text])[0]]
 
+
+# Request format is /predictBatch?SpareMeElement0=<encoded_string>&id_token=<token>
+@app.route('/predictBatch', methods=['GET'])
+def predict_batch():
+    # extract arguments from get request
+    try:
+        id_token = request.args['id_token']
+    except KeyError:
+        return "Error: must specify at least one SpareMeElement<index> arg and id_token arg when making a GET to " \
+               "/predictBatch "
+
+    try:
+        # Verify the ID token while checking if the token is revoked by
+        # passing check_revoked=True
+        decoded_token = auth.verify_id_token(id_token, check_revoked=True)
+        # Token is valid and not revoked
+        uid = decoded_token['uid']
+    except ValueError:
+        return "Error: token malformed"
+    except auth.AuthError as exc:
+        if exc.code == 'ID_TOKEN_REVOKED':
+            return "Error: token revoked; user must reauthenticate or sign out"
+        else:
+            return "Error: token invalid"
+
+    # get the actual user from the uid
+    user = auth.get_user(uid)
+
+    # control access using custom claim
+    if user.custom_claims is None or user.custom_claims.get('admin') is not True:
+        # Prevent access to admin resource.
+        return "Error: user is not allowed to access this resource"
+
+    # verify prediction pre-requisites
+    # if len(cats) < 2:
+    if len(cats) < 2 or any(i < 3 for i in Counter(target).values()):
+        return "Error: not enough data to predict"
+
+    predictions = {}
+
+    # predict categories of given text
+    for key in request.args:
```

Comment:
This iteration includes the auth token, which probably won't hurt anything, but we should probably not attempt to classify it regardless.

Reply:
what if I'm offended by authentication tho 

Agreed. I'll strip it out before merging